File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    var answer = '';
    
    let mode = 0;
    
    for (let idx = 0; idx < code.length; idx++) {
        if (code[idx] == '1') {
            mode == 0 ? mode = 1 : mode = 0
            
            console.log(mode)
        } else if (mode == 0 && idx % 2 == 0) {
            answer += code[idx]
            
            console.log(code[idx])
        } else if (mode == 1 && idx % 2 != 0) {
            answer += code[idx]
            
            console.log(code[idx])
        }
    }
    
    if (answer == '') {
        answer = "EMPTY"
    }
    
    return answer;
}

File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        int mode = 0;
        char tmp;
        
        for (int i = 0; i < code.length(); i++) {
            tmp = code.charAt(i);
            
            if (tmp == '1') {
                mode = (mode + 1) % 2;
            } else {
                if (mode == 1 && i % 2 == 1) {
                    answer += tmp;
                } else if (mode == 0 && i % 2 == 0) {
                    answer += tmp;
                }
            }
        }
        
        if (answer.length() == 0) {
            answer = "EMPTY";
        }
        
        return answer;
    }
}